File: lib/test/structuredfield/parseDate.test.ts
import assert from 'node:assert';
import test from 'node:test';
import { parseDate } from '../../src/structuredfield/parse/parseDate.js';

test('parseDate', () => {
	assert.deepStrictEqual(parseDate(`@1659578233`), { value: new Date(1659578233000), src: `` });
	assert.deepStrictEqual(parseDate(`@-1659578233`), { value: new Date('1917-05-30 22:02:47Z'), src: `` });
	assert.throws(() => parseDate(``), /failed to parse "" as Date/);
});

File: python/src/test/structuredfield/parseDate_test.py
import unittest
from datetime import datetime

from ...structuredfield.parse.parseDate import parseDate
from ...structuredfield.parse.ParsedValue import ParsedValue
from ...structuredfield.SfException import SfException


class TestDecodeSfItem(unittest.TestCase):
    def test_parseDate(self):
        self.assertEqual(
            parseDate("@1659578233"),
            ParsedValue(datetime.fromtimestamp(1659578233), ""),
        )
        self.assertEqual(
            parseDate("@-1659578233"),
            ParsedValue(datetime.fromtimestamp(-1659578233), ""),
        )
        self.assertRaises(SfException, parseDate, "")
